src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -180,22 +170,21 @@ public void notifyError(Throwable error)
     public void notifyProgress(String message)
     {
         logger.info(message);
-        fireProgressEvent(new ProgressEvent(ProgressEventType.PROGRESS,
-                                            progressCounter.incrementAndGet(),
-                                            totalProgress,
-                                            message));
+        fireProgressEvent(state.jmxEvent(ProgressEventType.PROGRESS, message));
     }
 
     private void skip(String msg)
     {
-        notification("Repair " + parentSession + " skipped: " + msg);
+        state.phase.skip(msg);
+        notification("Repair " + state.id + " skipped: " + msg);
         success(msg);
     }
 
     private void success(String msg)
     {
-        fireProgressEvent(new ProgressEvent(ProgressEventType.SUCCESS, progressCounter.get(), totalProgress, msg));
-        ActiveRepairService.instance.recordRepairStatus(cmd, ActiveRepairService.ParentRepairStatus.COMPLETED,
+        state.phase.success(msg);
+        fireProgressEvent(state.jmxEvent(ProgressEventType.SUCCESS, msg));
```

Comment:
not sure why `state` needs to produce `jmxEvent`. Can we keep it as state tracking only?

Reply:
moved this back to `RepairRunnable`